hazelcast/src/main/java/com/hazelcast/map/impl/operation/DeleteOperation.java

```diff
@@ -58,7 +58,7 @@ public void applyState(State state) {
 
     @Override
     public Object getResponse() {
-        return success;
+        return null;
```

Comment:
is this intentional change?

Reply:
Yes, it was intentional as written in the comments above. It seems we don't need to return a response here, we were not using the response

Reply:
ok makes sense to remove then.

Having a short inline comment for the reasoning of null usage here for future reference will be good.

Reply:
@ahmetmircik I have committed https://github.com/hazelcast/hazelcast/pull/23509/commits/8b6a1222170860e2de673a61cf204bf05a0bd12f, is this enough?

Reply:
Sure returning null is an option but just want to note my expectation here when i think myself as api user:

Returning a true/false seems more expected api behavior from user perspective. Yes, so far we return nothing to user for sync delete but for async variant, expectations changed and response is passed via callback, i prefer to see the result of api call instead of null.

But it can be only me expecting this, so both case can be ok.

Reply:
For the member proxy, that is easy to do. It will be a bit problematic for the client-side API. We would need to add the response parameter to the delete codec (that is fine and doable). However, the problem is that when the new client is connected to the old server (pre-5.3), we won't get the response, so we would not have a way of knowing whether it is true or false. If you think it is OK to always return false here (with a bit of documentation), we can do it if everyone agrees on it

Reply:
i'm ok with the approach you proposed. Maybe pre-5.3 can pass null as a response instead of false.

Reply:
Hmm, that might be a better approach. @vladykin do you want to work on this yourself? if not, I can work on the required protocol/API changes and push them on top of your commits. If you want to do it, basically, you will need to add a new response parameter to this https://github.com/hazelcast/hazelcast-client-protocol/blob/master/protocol-definitions/Map.yaml#L355 (with the since value of 2.6), and update the respective message task and APIs accordingly

Reply:
I'm going to take a look

Reply:
Took a look and now I have several questions to clarify.

1. If I understood the discussion above correctly, the idea is to make the `Delete` operation return true or false, depending on the success of the deletion. Basically, revert my change in the `DeleteOperation.getResponse()`, and instead make the boolean return official by describing it in the protocol spec, something like this. Is that correct?
```yaml
    response:
      params:
        - name: response
          type: boolean
          nullable: true
          since: 2.6
          doc: |
            Returns true if the key exists and removed, otherwise returns false.
```
2. Should the return value of `IMap.delete()` and `IMap.deleteAsync()` in the Java API be changed from void to boolean? 
3. Looks like Delete operation doesn't always know if the value was actually deleted. In the current implementation, if the value is not cached in memory, but exists in the MapStore, then the return value is `false`. Is that ok?
4. That's a separate git repository. What's the proper way to make related changes to two repositories?

Reply:
1. Yes, correct. Just make the `nullable: false`please, others look good.
2. Yes.
3. @ahmetmircik what do you think?
4. You do your changes in the hazelcast-client-protocol locally, generate codecs and add them to this PR. Then, you create a pull request on the hazelcast-client-protocol repo. After merging this PR, we will merge the protocol PR as well.

Reply:
Regarding 3, we can keep what we have currently, if key is in memory and deletion is successful, return true otherwise return false.

Reply:
AFAIK, changing the return type of an existing API method [is not a binary-compatible change](https://www.morling.dev/blog/refining-return-type-java-methods-without-breaking-backwards-compatibility/). That means, every dependent piece of code, that uses `IMap.delete()`, will break, when switched to the new version of `hazelcast.jar`. To avoid breaking, the code will have to be recompiled from source.

Is that acceptable? What are the backward compatibility requirements to the Hazelcast library?

Reply:
No, we cannot change the return type of the `void delete()` method as you said. We were talking about making the new API return a `CompletionStage<Boolean>`, instead of `CompletionStage<Void>`

Reply:
Done. Regenerated Java code with the updated protocol (1st commit) and added `CompletionStage<Boolean> deleteAsync(K key)` (2nd commit).

The build was failing for me locally. Enforcer plugin complained on the bytecode version in netty 4.1.90. I had to revert the recent bump of netty version to make the build pass. Hope someone will soon land a proper fix for this to the master branch, so I can remove this hack from the PR